pkg/build/util/generate.go

```diff
@@ -30,3 +32,43 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 		},
 	}
 }
+
+// SubstituteImageReferences replaces references to an image with a new value
+func SubstituteImageReferences(build *api.Build, oldImage string, newImage string) {
+	switch {
+	case build.Parameters.Strategy.Type == api.DockerBuildStrategyType &&
+		build.Parameters.Strategy.DockerStrategy != nil &&
```

Comment:
it's not invalid according to our current rules.
https://github.com/openshift/origin/blob/master/pkg/build/api/validation/validation.go#L102

Since there are no mandatory fields on DockerStrategy, it doesn't make sense to require one.